wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriter.scala

```diff
@@ -0,0 +1,450 @@
+      case a: IdentifierLookup => a.identifier
+      case a: IdentifierMemberAccess => a.toWdl
+      case a: ExpressionMemberAccess => s"${expressionElementWriter.toWdl(a.expression)}.${a.memberAccessTail.toList.mkString(".")}"
+      case a: IndexAccess => s"${expressionElementWriter.toWdl(a.expressionElement)}[${expressionElementWriter.toWdl(a.index)}]"
+    }
+  }
+
+  implicit val unaryOperationWriter: WdlWriter[UnaryOperation] = new WdlWriter[UnaryOperation] {
+    override def toWdl(a: UnaryOperation): String = a match {
+      case a: LogicalNot    => s"!(${a.argument.toWdl})"
+      case a: UnaryNegation => "-" + a.argument.toWdl
+      case a: UnaryPlus     => "+" + a.argument.toWdl
+    }
+  }
+
+  implicit val identifierMemberAccessWriter: WdlWriter[IdentifierMemberAccess] = new WdlWriter[IdentifierMemberAccess] {
+    override def toWdl(a: IdentifierMemberAccess): String = {
+      s"${a.first}.${a.second}" + (if (a.memberAccessTail.nonEmpty) {
+        "." + a.memberAccessTail.mkString(".")
+      } else {
+        ""
+      })
+    }
+  }
+
+  implicit val binaryOperationWriter: WdlWriter[BinaryOperation] = new WdlWriter[BinaryOperation] {
+    override def toWdl(a: BinaryOperation) = {
+      def sandwich(opString: String) = s"${a.left.toWdl} $opString ${a.right.toWdl}"
+
+      a match {
+        case _: LogicalOr           => sandwich("||")
+        case _: LogicalAnd          => sandwich("&&")
+        case _: Equals              => sandwich("==")
+        case _: NotEquals           => sandwich("!=")
+        case _: LessThan            => sandwich("< ")
+        case _: LessThanOrEquals    => sandwich("<=")
+        case _: GreaterThan         => sandwich("> ")
+        case _: GreaterThanOrEquals => sandwich(">=")
+        case _: Add                 => sandwich("+")
+        case _: Subtract            => sandwich("-")
+        case _: Multiply            => sandwich("*")
+        case _: Divide              => sandwich("/")
+        case _: Remainder           => sandwich("%")
+      }
+    }
+  }
+
+  implicit val graphElementWriter: WdlWriter[WorkflowGraphElement] = new WdlWriter[WorkflowGraphElement] {
+    override def toWdl(a: WorkflowGraphElement) = a match {
+      case a: CallElement => a.toWdl
+      case a: IntermediateValueDeclarationElement => a.toWdl
+      case a: OutputDeclarationElement => a.toWdl
+      case a: InputDeclarationElement => a.toWdl
+      case a: IfElement => a.toWdl
+      case a: ScatterElement => a.toWdl
+    }
+  }
+
+  implicit val scatterElementWriter: WdlWriter[ScatterElement] = new WdlWriter[ScatterElement] {
+    override def toWdl(a: ScatterElement): String =
+      s"""scatter (${a.scatterVariableName} in ${a.scatterExpression.toWdl}) {
+         |${indentAndCombine(a.graphElements.map(_.toWdl))}}""".stripMargin
+  }
+
+  implicit val callBodyElement: WdlWriter[CallBodyElement] = new WdlWriter[CallBodyElement] {
+    override def toWdl(a: CallBodyElement): String = {
```

Comment:
ToL: There's a new short-form syntax for these in scala 2.12 (which we can now use - hooray!). I'll let you decide whether you want to retype everything just for the sake of saving a few characters:
```scala
implicit val callBodyElement: WdlWriter[CallBodyElement] = a => {
  // method contents
}
```

Reply:
Thanks, good to know - in this particular case, I like having `toWdlV1` written out explicitly so people know what method we're implementing, the shortening seems to necessitate making it anonymous